### PR TITLE
ALMA: Use astroquery's download tools

### PR DESCRIPTION
**Pull request by keflavich**
The ALMA query tool was using straight `_request`s to get data, which is not great b/c those are not streamed, and individual ALMA files can exceed 100s of GB - too much for memory.

This PR refactors the Alma download tool to use astroquery's streamed download.

However, both variants are currently failing, so this is also an Issue: @andamian MWE:

```python
from astroquery.alma import Alma

result = Alma.query({'project_code':'2013.1.01365.S'})
uids = np.unique(result['member_ous_uid'])
files = Alma.get_data_info(uids)
savedir='/big/data/path/'
Alma.cache_dir=savedir

# this is a big one that fails (not clear why yet - OOM kill)
Alma.download_files([files['access_url'][3]])
```

### CODE REVIEW

**Comment by keflavich** [comment]
OK, running a quick test now, this seems to work - with `logLevel=5`, I get, e.g.:
```python
DEBUG: _response_hook_kwargs = {'timeout': 60, 'verify': True, 'proxies': OrderedDict(), 'stream': True, 'cert': None} [astroquery.query]
TRACE: HTTP response
        -----------------------------------------
        206 Partial Content https://almascience.nrao.edu/dataPortal/uid___A002_X8a951a_X69b
        Date: Wed, 17 Mar 2021 20:34:17 GMT
        Server: Apache/2.4.6 (Red Hat Enterprise Linux)
        Content-Disposition: attachment; filename=uid___A002_X8a951a_X69b.asdm.sdm.tar
        Content-Range: bytes 13565952-31660634111/31660634112
        Accept-Ranges: bytes
        X-Content-Type-Options: nosniff
        X-XSS-Protection: 1; mode=block
        Cache-Control: no-cache, no-store, max-age=0, must-revalidate
        Pragma: no-cache
        Expires: 0
        X-Frame-Options: DENY
        Content-Type: application/tar
        Content-Length: 31647068160
        Keep-Alive: timeout=5, max=100
        Connection: Keep-Alive
        Streaming Data
        ----------------------------------------- [astroquery.query]
```

so now even with the lowest-level debugging, the data won't be loaded from a streamed request

**Comment by keflavich** [comment]
The logging bugfix is pushed to #2021, and I rebased this against #2021 - that should be merged first.

**Comment by bsipocz** [comment]
I believe this requires a rebase as #2021 is now merged

**Comment by keflavich** [comment]
rebased

**Comment by bsipocz** [comment]
Overall it looks good to me, but would prefer to leave the formal approval to Adrian.

**Comment by andamian** [comment]
> Overall it looks good to me, but would prefer to leave the formal approval to Adrian.

It looks good to me @keflavich

**Comment by bsipocz** [comment]
Thanks for the confirmation, in that case I go ahead and merge this now.

File: astroquery/alma/core.py [hunk 1]
```diff
@@ -708,22 +708,47 @@ def download_files(self, files, savedir=None, cache=True,
...
+            try:
+                filename = re.search("filename=(.*)",
+                                     check_filename.headers['Content-Disposition']).groups()[0]
+            except KeyError:
+                log.info(f"Unable to find filename for {fileLink}  "
+                         "(missing Content-Disposition in header).  "
+                         "Skipping to next file.")
+
+            if savedir is not None:
+                filename = os.path.join(savedir,
```
Comment (bsipocz) [inline astroquery/alma/core.py:739]: Too bad, I didn't spot this:

`E               UnboundLocalError: local variable 'filename' referenced before assignment`

Comment (bsipocz) [reply astroquery/alma/core.py:739]: Hmm, the mocks in the tests also need some adjustments. Take away message, I should remote test PRs before merge :)

Comment (keflavich) [reply astroquery/alma/core.py:739]: I see.  This is an actual issue I can refer upstream - @andamian if you are able to chime in, it appears that some of the data sets do not have 'filename' metadata in the `Content-Disposition` headers.  Is this an archive bug I should raise via helpdesk?

Comment (keflavich) [reply astroquery/alma/core.py:739]: see https://github.com/astropy/astroquery/pull/2098

Comment (bsipocz) [reply astroquery/alma/core.py:739]: No, it's not the data, but the mock in the test.

Comment (bsipocz) [reply astroquery/alma/core.py:739]: (adding an easy fix there then messes up the assert that it has to bed triggered twice, and there are more complication with the second failing test)

Comment (keflavich) [reply astroquery/alma/core.py:739]: OK, it might not fix the failing test, but this fix _is_ necessary.

**Pull request merged**